static/skywire-manager-src/src/app/components/vpn/pages/vpn-server-list/vpn-server-list.component.ts

```diff
@@ -0,0 +1,877 @@
+import GeneralUtils from 'src/app/utils/generalUtils';
+import { EnterVpnServerPasswordComponent } from './enter-vpn-server-password/enter-vpn-server-password.component';
+
+/**
+ * Server lists VpnServerListComponent can show.
+ */
+export enum Lists {
+  /**
+   * List of public servers obtained from the discovery service.
+   */
+  Public = 'public',
+  /**
+   * List of the last servers to which the app has been connected.
+   */
+  History = 'history',
+  /**
+   * List with the favorite servers.
+   */
+  Favorites = 'favorites',
+  /**
+   * List with the blocked servers.
+   */
+  Blocked = 'blocked',
+}
+
+/**
+ * Data of a server for being shown in the list. Combines the data provided by the discovery
+ * service and the data saved locally.
+ */
+interface VpnServerForList {
+  /**
+   * 2 letter code of the country the server is in.
+   */
+  countryCode: string;
+  /**
+   * Sever name, obtained from the discovery service.
+   */
+  name: string;
+  /**
+   * Custom name set by the user.
+   */
+  customName: string;
+  /**
+   * Location of the server, obtained from the discovery service.
+   */
+  location: string;
+  /**
+   * Public key.
+   */
+  pk: string;
+  /**
+   * Current congestion of the server, obtained from the discovery service.
+   */
+  congestion?: number;
+  /**
+   * Rating of the congestion the server normally has, obtained from the discovery service.
+   */
+  congestionRating?: Ratings;
+  /**
+   * Latency of the server, obtained from the discovery service.
+   */
+  latency?: number;
+  /**
+   * Rating of the latency the server normally has, obtained from the discovery service.
+   */
+  latencyRating?: Ratings;
+  /**
+   * Hops needed for reaching the server.
+   */
+  hops?: number;
+  /**
+   * Note with information about the server, obtained from the discovery service.
+   */
+  note: string;
+  /**
+   * Personal note added by the user.
+   */
+  personalNote: string;
+  /**
+   * Last moment in which the VPN was connected to the server.
+   */
+  lastUsed?: number;
+  /**
+   * If the server is in the history of recently used servers.
+   */
+  inHistory?: boolean;
+  /**
+   * Special condition the server may have.
+   */
+  flag?: ServerFlags;
+  /**
+   * If the last time the server was used it was used with a password.
+   */
+  usedWithPassword?: boolean;
+  /**
+   * If the server was entered manually, at least one time.
+   */
+  enteredManually?: boolean;
+
+  /**
+   * Original VpnServer instance used for creating this object. Is not set if the objet was not
+   * created using a VpnServer instance.
+   */
+  originalDiscoveryData?: VpnServer;
+  /**
+   * Original LocalServerData instance used for creating this object. Is not set if the objet
+   * was not created using a LocalServerData instance.
+   */
+  originalLocalData?: LocalServerData;
+}
+
+/**
+ * Page for showing the vpn server lists.
+ */
+@Component({
+  selector: 'app-vpn-server-list',
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/skycoin/skywire/pullRequest?prid=7163595)
```suggestion
  selector: "app-vpn-server-list",
```